cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/event/EventBus.java

```diff
@@ -74,12 +78,29 @@ public void post(Object event) {
      * {@link com.cloudant.sync.event.Subscribe}.
      * </p>
      *
+     * <p>
+     *     The following restrictions are placed on the subscriber instance.
+     * </p>
+     * <ul>
+     *    <li>The class of the subscriber <strong>must</strong> be public</li>
+     *    <li>Methods annotated with <code>@Subscribe</code> <strong>must</strong> be public</li>
+     *    <li>Methods annotated with <code>@Subscribe</code> <strong>must</strong> have exactly <strong>one</strong> parameter</li>
+     * </ul>
+     *
      * @param object the instance that will be notified when an event is posted
+     * @throws IllegalArgumentException if the subscribing class does not adhere to the restrictions listed.
+     *
      */
     public void register(Object object) {
 
         Class<?> listenerClass = object.getClass();
 
+        if(!Modifier.isPublic(listenerClass.getModifiers())){
+            throw new IllegalArgumentException(String.format(Locale.ENGLISH,
+                    "Event Subscriber class %s needs to be public",
+                    listenerClass.isAnonymousClass() ? "Anonymous" : listenerClass.getCanonicalName()));
```

Comment:
Why not just use `listenerClass.getName()` for these? It would avoid the need to do the anonymous check because an anonymous class will be returned with a `$n` reference which will make it easier for people to identify the offending class.